main.nf

```diff
@@ -344,15 +420,15 @@ if (params.bam) {
     maxRetries 1
 
     input:
-    ksize
-    molecule
-    log2_sketch_size
+    each ksize from ksizes
+    each molecule from molecules
+    each log2_sketch_size from log2_sketch_sizes
     file(barcodes_file) from barcodes_ch
     set sample_id, file(bam) from bam_ch
     file(rename_10x_barcodes) from rename_10x_barcodes_ch
 
     output:
-    set val(sample_id), file("*.fasta") into reads_ch
+    set val(sample_id), file("*.fasta") into cell_barcode_fasta_ch
```

Comment:
this was intentional! I don't wanna compute and process all ksizes, molecule types for all permutations for every bam. I want to do it once on the first molecule and just do sourmash compute on the fastas formed later, please change it back! reads_ch is the channel that is input to the fasta files for the next compute function. please change it back